File: FrontEnd/src/services/categoryService.js
import axiosClient from "../lib/axiosClient";

const BASE_URL = "/api/categories";

export const getAllCategories = async (page = 0, size = 10, keyword = "") => {
  const params = { page, size };
  if (keyword) params.keyword = keyword;
  const response = await axiosClient.get(`${BASE_URL}/getAllCategories`, { params });
  return response.data;
};

export const createCategory = async (data) => {
  const response = await axiosClient.post(`${BASE_URL}/create`, data);
  return response.data;
};

export const updateCategory = async (id, data) => {
  const response = await axiosClient.put(`${BASE_URL}/update/${id}`, data);
  return response.data;
};

export const deleteCategory = async (id) => {
  const response = await axiosClient.delete(`${BASE_URL}/delete/${id}`);
  return response.data;
};

File: BackEnd/src/main/java/com/Library/lmsproject/service/CategoryService.java
package com.library.lmsproject.service;

import com.library.lmsproject.dto.request.CreateCategoryRequestDTO;
import com.library.lmsproject.dto.request.UpdateCategoryRequestDTO;
import com.library.lmsproject.dto.response.CategoryResponseDTO;
import org.springframework.data.domain.Page;

public interface CategoryService {
    CategoryResponseDTO createCategory(CreateCategoryRequestDTO request);

    Page<CategoryResponseDTO> getAllCategories(int page, int size, String keyword);

    Boolean deleteCategory(Long categoryId);

    CategoryResponseDTO updateCategory(Long categoryId,
                                       UpdateCategoryRequestDTO request);
    CategoryResponseDTO getCategoryById(Long categoryId);
}
